src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/OffenderSearchService.kt

```diff
@@ -16,36 +16,43 @@ class OffenderSearchService(
   @Qualifier("communityApiClientUserEnhanced") private val communityApiClient: CommunityApiClient
 ) {
   suspend fun search(crn: String): List<SearchByCrnResponse> {
-    val request = OffenderSearchByPhraseRequest(
-      phrase = crn
-    )
-    val apiResponse = getValue(offenderSearchApiClient.searchOffenderByPhrase(request))?.content
-
-    return apiResponse?.map {
-      var name = "${it.firstName} ${it.surname}"
-      var excluded: Boolean? = null
-      var restricted: Boolean? = null
-
-      // Check whether an empty name is genuinely due to a restriction or exclusion
-      if (it.firstName == null && it.surname == null) {
-        val userAccessResponse = getValue(communityApiClient.getUserAccess(crn))
+    val userAccessResponse = getValue(communityApiClient.getUserAccess(crn))
+    return if (true == userAccessResponse?.userRestricted || true == userAccessResponse?.userExcluded) {
+      listOf(
+        SearchByCrnResponse(
+          crn = crn,
+          userExcluded = userAccessResponse.userExcluded,
+          userRestricted = userAccessResponse.userRestricted
+        )
+      )
+    } else {
+      val request = OffenderSearchByPhraseRequest(
+        phrase = crn
+      )
+      val apiResponse = getValue(offenderSearchApiClient.searchOffenderByPhrase(request))?.content
+      apiResponse?.map {
+        var name = "${it.firstName} ${it.surname}"
+        var excluded: Boolean? = null
+        var restricted: Boolean? = null
 
-        if (true == userAccessResponse?.userRestricted || true == userAccessResponse?.userExcluded) {
-          excluded = userAccessResponse.userExcluded
-          restricted = userAccessResponse.userRestricted
-        } else {
-          name = "No name available"
+        // Check whether an empty name is genuinely due to a restriction or exclusion
+        if (it.firstName == null && it.surname == null) {
+          if (true == userAccessResponse?.userRestricted || true == userAccessResponse?.userExcluded) {
```

Comment:
This block of code doesn't look quite right now. The block can never be true as the code on line 20 always returns first. 

When I put the original code in for this, I was trying to stop it hitting the `communityApiClient.getUserAccess()` call for every single request, partly because it isn't future proof if we expand search in the future for search terms that are more broad than crn. For example, searching by name could return 100s or 1000s of cases and you would need to do an API call to `getUserAccess()` for every one of these which could have a significant effect on performance.

Reply:
> This block of code doesn't look quite right now. The block can never be true as the code on line 20 always returns 
> first.

That's true!! 

Then again the original code AFAIK couldn't have worked either. The comment says -
` // Check whether an empty name is genuinely due to a restriction or exclusion`
This assumes that the name is retrievable if there is a restriction/exclusion. As far as I can see it isn't.
If there is a restriction an attempt to call to the probation-search API will result in a 403.

I think the solution here is to remove the null name check.

> When I put the original code in for this, I was trying to stop it hitting the communityApiClient.getUserAccess() call for every single request, partly because it isn't future proof

That might be prudent. My opinion here would be to stick to the Extreme Programming YAGNI principle, and cross that bridge when we come to it.

I could be wrong, so correct me if I am! I'm also not married to my proposals, so feel free to disagree!

Hmm, this feels more like a huddle conversation :)
Anyway, we can have a call about it soon!

Reply:
Addressed as discussed.